src/Agent.Plugins/PipelineCache/TarUtils.cs

```diff
@@ -17,8 +17,8 @@ namespace Agent.Plugins.PipelineCache
     public static class TarUtils
     {
         private readonly static bool isWindows = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
-
-        private const string archiveFileName = "archive.tar";
+        private const string archive = "archive.tar";
+        private static string archiveFileName = $"{Guid.NewGuid().ToString("N")}_{archive}";
```

Comment:
>private static string archiveFileName [](start = 8, length = 37)

We can't use static string here because it will only create one archiveFileName in the process' lifetime. It needs to be a CreateArchiveFileName function where it can create a new archive file name.

Reply:
Done.